### PR TITLE
SelectBox: fix bug with excess selections when selected item is not on the first page (T1025148)

### PR DESCRIPTION
**Pull request by Zedwag**

### CODE REVIEW

File: testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js [hunk 1]
```diff
@@ -3974,6 +3974,41 @@ QUnit.module('Async tests', {}, () => {
...
+        const selectBox = $selectBox.dxSelectBox('instance');
+        selectBox.open();
+        const $list = $(selectBox.content()).find('.dx-list');
+
+        setTimeout(() => {
+            const $selectedItems = $list.find(toSelector(LIST_ITEM_SELECTED_CLASS));
+
+            assert.strictEqual($selectedItems.length, 0, 'no items are selected');
+            done();
+        }, 0);
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js]: I think it's better to use more readable settimeout and sinon.useFakeTimers for sure test is stable

File: js/ui/selection/selection.strategy.standard.js [hunk 2]
```diff
@@ -390,7 +390,7 @@ export default SelectionStrategy.inherit({
     _isItemSelectionInProgress: function(key, checkPending) {
         const shouldCheckPending = checkPending && this._lastRequestData && this._requestInProgress();
         if(shouldCheckPending) {
-            return this._lastRequestData.addedItems?.indexOf(key) !== -1;
+            return this._lastRequestData.addedItems && this._lastRequestData.addedItems.indexOf(key) !== -1;
```
Comment (DokaRus) [inline js/ui/selection/selection.strategy.standard.js]:
```suggestion
            const addedItems = this._lastRequestData.addedItems ?? [];
            return addedItems.includes(key);
```

File: testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js (hunk 1 above)
Comment (Zedwag) [reply testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js]: done

File: js/ui/selection/selection.strategy.standard.js (hunk 2 above)
Comment (Zedwag) [reply js/ui/selection/selection.strategy.standard.js]: done

**Review by DokaRus — DISMISSED** [review]

**Comment by ksercs** [comment]
Let's name the PR more specific. Which bug with selection?

The suffix with ticket id does not comply with the our PR naming convention

File: testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js [hunk 3]
```diff
@@ -3974,6 +3974,46 @@ QUnit.module('Async tests', {}, () => {
...
+            }).dxSelectBox('instance');
+            const $list = $(selectBox.content()).find('.dx-list');
+
+            this.clock.tick(50);
+
+            const $selectedItems = $list.find(toSelector(LIST_ITEM_SELECTED_CLASS));
+
+            assert.strictEqual($selectedItems.length, 0, 'no items are selected');
+        } catch(e) {
+            assert.ok(false, 'Exception: ' + e);
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js]: do we expect some kind of error in this test?

According to the test name, we should check only items selection. If we wanna to check that no errors is raised in some scenario, it's a good idea to write a separate test with a specific name

Comment (Zedwag) [reply testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js]: done

**Review by alexanderPolosatov — APPROVED** [review]

**Review by alexander-kotov-dx — APPROVED** [review]

**Pull request merged**